appcues/src/main/java/com/appcues/Appcues.kt

```diff
@@ -139,7 +140,12 @@ public class Appcues internal constructor(internal val scope: AppcuesScope) {
 
         appcuesCoroutineScope.launch {
             analyticsTracker.analyticsFlow.collect {
-                analyticsPublisher.publish(analyticsListener, it)
+                try {
+                    analyticsPublisher.publish(analyticsListener, it)
+                } catch (_: Exception) {
+                    // ignore any exception that occurs in client app code that is observing analytics,
+                    // so we always continue collecting on the analyticsFlow
+                }
```

Comment:
this is a small enhancement we talked about a little while back - passing a call over to the customer listener is going into unknown code, so we should just safeguard for any exception and carry on, so we don't kill this coroutine and stop the analytics flow.

Reply:
any idea how we could track if exceptions are getting thrown here? would be nice if at least the debugger was notified that. something happened here so if they see it while using with the debugger open they can report it and we can patch it up? (for later PR)

Reply:
yeah great call, we can at least send it through Logcues I assume

Reply:
Simple update in this commit https://github.com/appcues/appcues-android-sdk/pull/648/commits/701ec8cf7e55bdc37336849703effb0119e8a968

So they show up in the detailed log like this:

![Screenshot 2024-10-21 at 10 08 13 AM](https://github.com/user-attachments/assets/291bf7b4-9378-4c58-8008-e70de4e3c627)

thanks